src/include/ipc/topology.h

```diff
@@ -64,6 +64,7 @@ struct sof_ipc_comp {
 	enum sof_comp_type type;
 	uint32_t pipeline_id;
 	uint32_t core;
+	uint8_t uuid[16];	/**< 16 Bytes UUID of the component */
```

Comment:
can we use the same a.b.c.d structure that @mmaka1 has already defined. This will need to be a new IPC struct too.

Reply:
Andy from the kernel side suggest not to use uuid_t or guid_t struct as ABI, see here: https://github.com/thesofproject/linux/pull/2090#discussion_r423313872

Reply:
use the FW structure here exactly.